test/e2e/lib/hooks/index.js

```diff
@@ -1,10 +1,50 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+
 /**
  * Internal dependencies
  */
-import { startVideo, stopVideo, takeScreenshot } from './video-recorder';
+import { buildHooks as buildVideoHooks } from './video-recorder';
+import { buildHooks as buildFramebufferHooks, getFreeDisplay } from './framebuffer';
+
+const isVideoEnabled = () => {
+	const video = config.has( 'useTestVideo' )
+		? config.get( 'useTestVideo' )
+		: process.env.TEST_VIDEO;
+	return video === 'true';
+};
+
+export const mochaHooks = async () => {
+	const hooks = {
+		afterAll: [
+			() => {
+				console.log( 'After all: root' );
+			},
+		],
+		beforeAll: [],
+		afterEach: [],
+	};
+
+	if ( isVideoEnabled() ) {
+		const displayNum = await getFreeDisplay();
+
+		const { startFramebuffer, stopFramebuffer, takeScreenshot } = buildFramebufferHooks(
+			displayNum
+		);
+		const { startVideoRecording, saveVideoRecording, stopVideoRecording } = buildVideoHooks(
```

Comment:
I haven't seen this syntax before. Do you have somewhere I can read up on this?

Reply:
It's called [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring)

Reply:
Sorry, I meant moreso in the sense of how you were building the hooks. I hadn’t seen hooks built this way in documentation.

Reply:
Ah I see what you mean. I don't know if this pattern has a name.

It's worth mentioning that at this point, those functions are not hooks, they haven't been attached to Mocha yet. So this is a function `buildFramebufferHooks` that returns an object with three functions `{ startFramebuffer, stopFramebuffer, takeScreenshot }`.

I'm using it so those three functions can have access to the same `displayNum` (defined here) and `xvfb` (defined in `framebuffer.js`), without exposing those values as globals.